### ISSUE TITLE
slow performance of LRN layer

### ISSUE THREAD

**Issue by hwnam831** [Lasagne/Lasagne#721]
I tried a simple alexnet model benchmark using LocalResponseNormalization2DLayer

The model looks like this
`layer = Conv2DDNNLayer(layer, 96, 11, stride=4, pad=2)
   # layer = LocalResponseNormalization2DLayer(layer, 0.0001, 2, 0.75, 5)
    layer = MaxPool2DDNNLayer(layer, 3, stride=2)

    layer = Conv2DDNNLayer(layer, 256, 3, pad=2)
    #layer = LocalResponseNormalization2DLayer(layer, 0.0001, 2, 0.75, 5)
    layer = MaxPool2DDNNLayer(layer, 3, stride=2)

    layer = Conv2DDNNLayer(layer, 384, 3, pad=1)

    layer = Conv2DDNNLayer(layer, 384, 3, pad=1)

    layer = Conv2DDNNLayer(layer, 256, 3, pad=1)
    layer = MaxPool2DDNNLayer(layer, 3, stride=2)

    layer = DenseLayer(layer, 4096)
    layer = DropoutLayer(layer, p=0.5)
    
    layer = DenseLayer(layer, 4096)
    layer = DropoutLayer(layer, p=0.5)
    
    layer = DenseLayer(layer, 4096)
	
    layer = DenseLayer(layer, 1000, nonlinearity=lasagne.nonlinearities.softmax)
`
This model takes approximately 300ms for each training batch
But when I uncomment the LRN part, it takes about 700ms.
I profiled the code with nvprof and found out that k_ielem_4 cuda kernel takes about 50% of gpu running time.

**Comment by benanne** [comment]
Unfortunately, this layer is implemented very inefficiently. Making it faster would probably require custom C and/or CUDA code. We have not made it a priority to optimise this because basically, nobody uses it anymore. It is only really useful when reproducing AlexNet nowadays.

A faster implementation of this operation is in fact available in Theano, through the pylearn2 cuda-convnet wrappers, but we never made the effort to wrap it in a Layer class. If you're interested in trying this, have a look at the existing cuda-convnet wrapper layers here: https://github.com/Lasagne/Lasagne/blob/master/lasagne/layers/cuda_convnet.py
The pylearn2 docs for the normalization op are here: http://deeplearning.net/software/pylearn2/library/alex.html#module-pylearn2.sandbox.cuda_convnet.response_norm

**Comment by nouiz** [comment]
The ideal would be to wrap the cuDNN implementation of that. But this would
take more time then wrapping the code in Pylearn2. But as said, it is low
in the priority list. So don't expect someone else to do it.
